front/pages/api/w/[wId]/templates/index.ts

```diff
@@ -0,0 +1,142 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import {
+  deleteTemplate,
+  getGensTemplates,
+  getTemplate,
+  updateTemplate,
+} from "@app/lib/api/gens";
+import { Authenticator, getSession, getUserFromSession } from "@app/lib/auth";
+import { GensTemplate } from "@app/lib/models";
+import { new_id } from "@app/lib/utils";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { GensTemplateType } from "@app/types/gens";
+
+export type GetTemplatesResponseBody = {
+  templates: GensTemplateType[];
+};
+
+export type PostTemplatesResponseBody = {
+  template: GensTemplateType;
+};
+export type UpdateTemplatesResponseBody = {
+  template: GensTemplateType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<GetTemplatesResponseBody | PostTemplatesResponseBody>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  const user = await getUserFromSession(session);
+  const isBuilder = auth.isBuilder();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  const body = req.body;
+  let template, result;
+  switch (req.method) {
+    case "GET":
+      const temp_data = await getGensTemplates(owner, user);
+      res.status(200).json({
+        templates: temp_data,
+      });
+      return;
+
+    case "POST":
```

Comment:
To be RESTFUL we have 2 solutions:
- One post here that creates the template and return it with a fresh sId, then the delete and the update is on /templates/[tId]/index.ts 
- Or we go with an upsert logic and we have just a POST and DELETE on /templates/[tId]/index.ts